### PR TITLE
Scheduler fix

### LINKED ISSUE #42: scheduler crashes if dirs don't exist

### PR DESCRIPTION
**Pull request by rpanah**
Scheduler now creates dirs if they don't exist. This closes #42.
@ben-jones, please review this.

### CODE REVIEW

**Comment by ben-jones** [comment]
@rpanah, I'm not sure that we want to create the directories if they don't exist. Currently, we create directories whenever the clients connect. If the client does not exist, the current code base will crash (it might be helpful to print an error message telling the user that the client does not exist). We don't want to create the folder, because that means that we are creating users without any of the attached database stuff.

tl;dr I think we should just print an error message and exit.

@rpanah, can you think of a use case where you would want the directory to be created? (this should already be handled by the server)

**Comment by rpanah** [comment]
@ben-jones, I would argue that we do want to create those dirs for users that were created before the feature was added. Additionally, the scheduler connects to the database (using `Clients`), to find clients. This means that the users are not created without the database (in fact, the scheduler does not "create" users at all).

**Comment by phillipa** [comment]
any progress over here?

**Comment by ben-jones** [comment]
I feel like creating those directories is going to create confusion in the future. The goal of the scheduler is to schedule experiments on specific machine, not to create folders associated with clients. I think that we should separate functionality here by putting the bug fix into the client creation code (I think this is already done) and printing an error message to users.

Based on your argument that we are going through the database (this doesn't have to happen, but whatever), I would be ok with creating the directories, but this isn't the best way to handle the problem.

**Comment by phillipa** [comment]
Abbas can we just do that?

**Comment by rpanah** [comment]
@phillipa, sure. Added new commit. @ben-jones, please review.
